core/kazoo_number_manager/src/knm_phone_number.erl

```diff
@@ -483,10 +485,12 @@ set_prev_assigned_to(N, PrevAssignedTo=?MATCH_ACCOUNT_RAW(_)) ->
 %% @doc
 %% @end
 %%--------------------------------------------------------------------
--spec used_by(knm_phone_number()) -> api_binary().
+-spec used_by(knm_phone_number()) -> api_ne_binary().
 used_by(#knm_phone_number{used_by=UsedBy}) -> UsedBy.
 
--spec set_used_by(knm_phone_number(), ne_binary()) -> knm_phone_number().
+-spec set_used_by(knm_phone_number(), api_ne_binary()) -> knm_phone_number().
+set_used_by(N, UsedBy='undefined') ->
+    N#knm_phone_number{used_by=UsedBy};
```

Comment:
Because `when UsedBy == 'undefined'; UsedBy = ?NE_BINARY ->` is an illegal guard.